### PR TITLE
Add Nix Studio OXALYS80 Keyboard

### PR DESCRIPTION
**Pull request by sendz**
## Description

Add Nix Studio OXALYS80 TKL PCB

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/nix_studio/oxalys80/config.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x6E78
+#define PRODUCT_ID      0x3830
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Nix Studio
+#define PRODUCT         oxalys80
+#define DESCRIPTION     QMK-programmable TKL PCB for OXALYS80
```
Comment (drashna) [inline keyboards/nix_studio/oxalys80/config.h:23]: This is no longer used
```suggestion
```

File: keyboards/nix_studio/oxalys80/config.h [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+#define CONFIG_H
+
+#include "config_common.h"
```
Comment (drashna) [inline keyboards/nix_studio/oxalys80/config.h:27]:
```suggestion
```

File: keyboards/nix_studio/oxalys80/config.h [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/* USB Device descriptor parameter */
```
Comment (drashna) [inline keyboards/nix_studio/oxalys80/config.h:17]:
```suggestion
#pragma once

#include "config_common.h"

/* USB Device descriptor parameter */
```

File: keyboards/nix_studio/oxalys80/keymaps/default/config.h [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/* Copyright 2021 Nix Studio
```
Comment (drashna) [inline keyboards/nix_studio/oxalys80/keymaps/default/config.h:1]: this file can/should be removed.

File: keyboards/nix_studio/oxalys80/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+}
+
+void matrix_scan_user(void) {
+
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
```
Comment (drashna) [inline keyboards/nix_studio/oxalys80/keymaps/default/keymap.c:48]:
```suggestion
```

File: keyboards/nix_studio/oxalys80/oxalys80.c [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+	led_set_user(usb_led);
+}
+
+bool led_update_kb(led_t led_state) {
+    if(led_update_user(led_state)) {
+        writePin(E6, !led_state.caps_lock);
+        writePin(B2, !led_state.scroll_lock);
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/nix_studio/oxalys80/oxalys80.c:52]:
```suggestion
```

File: keyboards/nix_studio/oxalys80/config.h [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+#define RGB_DI_PIN B3
+#ifdef RGB_DI_PIN
+#define RGBLIGHT_ANIMATIONS
+#define RGBLED_NUM 24
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#define RGBLIGHT_LIMIT_VAL 200
+#define RGBLIGHT_SLEEP // RGB will turn off when PC is put to sleep
+#endif
```
Comment (drashna) [inline keyboards/nix_studio/oxalys80/config.h:76]:
```suggestion
#endif

#define LED_CAPS_LOCK_PIN E6
#define LED_SCROLL_LOCK_PIN B2
#define LED_PIN_ON_STATE 0
```

File: keyboards/nix_studio/oxalys80/rules.mk [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality on B7 by default
+RGBLIGHT_ENABLE = yes       # Enable RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+EXTRAFLAGS += -flto
```
Comment (drashna) [inline keyboards/nix_studio/oxalys80/rules.mk:25]:
```suggestion
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
LTO_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/nix_studio/oxalys80/config.h [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x6E78
+#define PRODUCT_ID      0x3830
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Nix Studio
+#define PRODUCT         oxalys80
```
Comment (sendz) [inline keyboards/nix_studio/oxalys80/config.h:26]:
```suggestion
#define PRODUCT         OXALYS80
```

**Review by tzarc — APPROVED** [review]

**Pull request merged**